sandbox.go

```diff
@@ -674,24 +674,62 @@ func (sb *sandbox) SetKey(basePath string) error {
 	return nil
 }
 
-func (sb *sandbox) EnableService() error {
+func (sb *sandbox) EnableService() (err error) {
 	logrus.Debugf("EnableService %s START", sb.containerID)
+	var n *network
+	var epFromStore *endpoint
+	defer func() {
+		if err != nil {
+			sb.DisableService()
+		}
+	}()
 	for _, ep := range sb.getConnectedEndpoints() {
-		if ep.enableService(true) {
-			if err := ep.addServiceInfoToCluster(sb); err != nil {
-				ep.enableService(false)
-				return fmt.Errorf("could not update state for endpoint %s into cluster: %v", ep.Name(), err)
+		if !ep.isServiceEnabled() {
+			n, err = ep.getNetworkFromStore()
```

Comment:
What does the store add here? can we keep the original logic that uses the *Endpoint coming from the map?

Reply:
store doesnt add anything.  Iam following the convention on of operating on the store object